backend/manager/modules/common/src/main/java/org/ovirt/engine/core/common/utils/VmCommonUtils.java

```diff
@@ -20,6 +20,12 @@ public class VmCommonUtils {
      * @return true, if any CPUs are to be hotplugged, false otherwise
      */
     public static boolean isCpusToBeHotpluggedOrUnplugged(VM source, VM destination) {
+        return cpuSocketsChanged(source, destination)
+                && !source.getCpuPinningPolicy().isExclusive()
+                && !destination.getCpuPinningPolicy().isExclusive();
```

Comment:
so we won't be able to change the policy from, let's say, dedicated to none and plug CPUs?

Reply:
Not on the running VM, no. The VM would need to be restarted for the changes to be applied.

Reply:
reading it again, my question above was incorrect - I meant to change a running vm that is set with policy=none to policy=dedicated and plug CPUs

Reply:
Every change of cpu pinning policy requires VM reboot. And it is not possible to hot-plug CPUs when using exclusive pinning. Your example would break both conditions :-)

There is no special reason for blocking it, it is just scenario that cannot currently work. How would you imagine going from none to exclusive while running the VM? Would we add pinning at runtime? I do not think VDSM supports that either.

Reply:
let's say that I take a non-stateless vm with 1 cpu and change it to be stateless with 2 cpus, I'll be able to say that the change to the cpu topology should happen now and I have no option but to wait for the stateless property to change when the vm restart:
![cpu_stateless](https://user-images.githubusercontent.com/6020068/180805546-cbd79c68-ac4c-4e5c-bfac-4d39047a7381.png)

similarly, if I take a vm with 1 cpu and policy=none, and change it to 2 cpus and policy=dedicated then I realize that the change to the policy is like the change to the stateless property (and thus, it would be applied on next start) but I would still expect to be able to hot-plug that additional cpu. after restart we won't consider this additional cpu as a plugged one, it would be part of the topology and so the vm should run property after restart with 2 cpus and dedicated policy, no?

Reply:
I see, thanks for the explanation. I've updated the patch to allow your scenario.